### PR TITLE
enforce published user limit when no license

### PR DESCRIPTION
**Pull request by sqs**
The [pricing page](https://about.sourcegraph.com/pricing/) states the limit is 10 users. Previously, we had an unpublished "grace period" that allowed up to 20 users. This change makes it so the published limit (10 users) is actually enforced.

If an instance is currently in the "grace period" (i.e., it has between 11-20 user accounts), then it will not be able to create new user accounts. If a new user signs up (over 10 users) or a site admin tries to create a new user account (over 10 users), an error will be displayed:

> Unable to create user account: a Sourcegraph subscription is required to exceed %d users (this instance now has %d users). Contact Sourcegraph to learn more at https://about.sourcegraph.com/contact/sales.

(The `%d` are filled in by the actual numbers.)

Also, all users will see the following notice at the top of each page:

> This Sourcegraph instance has reached the limit for free users, and an admin must purchase a license or contact Sourcegraph for a free trial.

Previously, e.g., an instance with 15 user accounts could create more user accounts (up to 20) despite being over the limit. An instance with over 10 user accounts is not immediately disabled or locked out as a result of this change (doing so would be very bad).

This affects some trials and prospects. See https://docs.google.com/spreadsheets/d/1C7ih9-w6Y21TPFfM79RFnrHXioyQxn5k2Uh3metPFbw/edit#gid=0 for a list.

### CODE REVIEW

**Comment by sqs** [comment]
@kpsource @juliasourceress @dadlerj Please review the list of affected instances at docs.google.com/spreadsheets/d/1C7ih9-w6Y21TPFfM79RFnrHXioyQxn5k2Uh3metPFbw/edit#gid=0. I called out a few ones in gdoc comments for each of you. I would like to merge this by 11am PDT today so it will apply starting in version 3.15 (released April 20), so we don't need to wait another month to do this. (This is one step toward implementing the new pricing plans, which also cap free usage at 10 users.) It is very easy to undo this change between now and April 20 (and prevent it from taking effect for anyone ever) in case something unexpected comes up.

**Comment by dadlerj** [comment]
(Note for others that this is dependent on https://github.com/sourcegraph/about/pull/769)

@sqs in my mind, the 20-user limit actually tied well to the 10-active user limit, as we know that active users is always strictly lower than user accounts. So my only question with this change — why reduce the actual implemented count from 20 to 10 and have to deal with the pain of notifying the users on the list above? 

We are changing pricing to reflect a new structure in the coming days/weeks regardless, so I'd rather just wait for that to happen to avoid doing two changes, one shortly the other.

So basically my proposal is to change https://github.com/sourcegraph/about/pull/769 to reflect what we actually do now (i.e., say "20-user limit") instead.

**Comment by sqs** [comment]
Merging based on code review approval and Slack thread with @dadlerj (https://sourcegraph.slack.com/archives/C0B2RU51Q/p1586881689347600?thread_ts=1586879475.347000&cid=C0B2RU51Q). The affected instances can be reviewed separately by @kpsource and @juliasourceress (the change will not take effect until the instances upgrade, which they can do on April 20 at the earliest, ~6 days from now, so we have time).
